zulipterminal/ui_tools/boxes.py

```diff
@@ -104,9 +104,11 @@ def generic_autocomplete(self, text: str, state: int) -> Optional[str]:
             if text.startswith(prefix):
                 typeaheads = autocomplete_func(text, prefix)
                 reduced_typeaheads = typeaheads[:num_suggestions]
-                try:
+
+                if state in range(-len(reduced_typeaheads),
+                                  len(reduced_typeaheads)):
```

Comment:
I tried using `range` for a huge 20-digit number and it still works without any time delay.

Reply:
I used `range` because it's easier to write, than specifying bounds explicitly.
And the range is not `[-n , n]`, but rather `[-n, n)`. So I'd have to check for negative/positive states and compare those separately. This is a lot more concise.

Reply:
Re a large range, that likely depends on `state`?

Also, I was leaning towards the more explicit `state is not None and -n <= state < n`, which is pretty clear and involves no data-structure.

Reply:
Oh right, that can work too